src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes/HttpContextRefactor/HttpContextRefactorCodeFixProvider.cs

```diff
@@ -0,0 +1,297 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes
+{
+    public abstract class HttpContextRefactorCodeFixProvider<TInvocationNode, TArgument> : CodeFixProvider
+        where TInvocationNode : SyntaxNode
+        where TArgument : SyntaxNode
+    {
+        private const string DefaultArgumentName = "currentContext";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(HttpContextCurrentAnalyzer.DiagnosticId); }
+        }
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        private protected abstract TInvocationNode AddArgumentToInvocation(TInvocationNode invocationNode, TArgument argument);
+
+        private protected abstract bool IsEnclosedMethodOperation(IOperation operation);
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (root is null)
+            {
+                return;
+            }
+
+            var diagnostic = context.Diagnostics.First();
+            var semantic = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (semantic is null)
+            {
+                return;
+            }
+
+            // Find the type declaration identified by the diagnostic.
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+            if (semantic.GetOperation(node, context.CancellationToken) is not IPropertyReferenceOperation property)
+            {
+                return;
+            }
+
+            var methodOperation = GetEnclosingMethodOperation(property);
```

Comment:
we're in the code fixer. is this check still necessary? seems like something the analyzer should handle. if there is no enclosingMethodOperation then we wouldn't raise the diagnostic

Reply:
It's used to identify where to add parameters